From 7440dda3332f3c07032e95dd22f688fa1675c12c Mon Sep 17 00:00:00 2001
From: wang, biao <biao.wang@intel.com>
Date: Wed, 12 Dec 2012 17:10:30 +0800
Subject: Avoid removing audio command that some thread is waiting for status

BZ: 73035

When a client thread insert a comand to mAudioCommands, it try to remove
the same pending command to avoid reduplicative handling, but for the
command with true mWaitStatus, there was a thread waitting for it's
response, removing it would lead this thread never wakeup and lead into
anr.

Category: AOSP Improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: no, needs rework

Change-Id: Ic47c77418afcf9ce35a5c0e8db7abae6771c505a
Orig-Change-Id: I038464b888b0dcf91ade91cd6ac103be2348c487
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/AudioPolicyService.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/services/audioflinger/AudioPolicyService.cpp b/services/audioflinger/AudioPolicyService.cpp
index 9fc4309..86cd28b 100644
--- a/services/audioflinger/AudioPolicyService.cpp
+++ b/services/audioflinger/AudioPolicyService.cpp
@@ -992,6 +992,7 @@ void AudioPolicyService::AudioCommandThread::insertCommand_l(AudioCommand *comma
         // commands are sorted by increasing time stamp: no need to scan the rest of mAudioCommands
         if (command2->mTime <= command->mTime) break;
         if (command2->mCommand != command->mCommand) continue;
+        if (command2->mWaitStatus) continue;
 
         switch (command->mCommand) {
         case SET_PARAMETERS: {
-- 
1.7.4.1

